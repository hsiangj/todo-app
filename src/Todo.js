import React, {useState} from 'react';

const Todo = ({id, task, remove, edit }) => {
  const [editTask, setEditTask] = useState(task);
  const [isEditing, setIsEditing] = useState(false);

  const toggleEdit = () => {
    setIsEditing(!isEditing);
  }

  const handleChange = (e) => {
    setEditTask(e.target.value);
  }

  const handleEdit = (e) => {
    e.preventDefault();
    edit(id, editTask);
    setIsEditing(false);
  }

  let jsx = (
    <li>
      {task}
      &nbsp;
      <button onClick={toggleEdit}>Edit</button>
      <button onClick={()=>remove(id)}>X</button>
    </li>
  )

  if(isEditing){
    jsx = (
      <form onSubmit={handleEdit}>
        <input type="text" value={editTask} onChange={handleChange}/>
        <button>Save</button>
      </form>
    )
  }


  return jsx;
}

export default Todo;