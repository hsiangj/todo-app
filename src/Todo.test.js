import { render, fireEvent } from '@testing-library/react';
import Todo from './Todo';

it("renders without crashing", function(){
  render(<Todo />);
});

it("matches snapshot", function(){
  const {asFragment} = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when editing", function(){
  const {asFragment, getByText} = render(<Todo />);
  const editButton = getByText('Edit');
  fireEvent.click(editButton);
  expect(asFragment()).toMatchSnapshot();
});

it("runs edit function on form submit", function(){
  const editMock = jest.fn();
  const {getByText} = render(<Todo edit={editMock} />);
  const editButton = getByText('Edit');
  fireEvent.click(editButton);
  const saveButton = getByText('Save');
  fireEvent.click(saveButton);
  expect(editMock).toHaveBeenCalled();
});

it("runs remove function on form button click", function(){
  const removeMock = jest.fn();
  const {getByText} = render(<Todo remove={removeMock} />);
  const removeButton = getByText('X');
  fireEvent.click(removeButton);
  expect(removeMock).toHaveBeenCalled();
});