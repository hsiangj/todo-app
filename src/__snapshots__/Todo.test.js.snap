// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li>
    Â 
    <button>
      Edit
    </button>
    <button>
      X
    </button>
  </li>
</DocumentFragment>
`;

exports[`matches snapshot when editing 1`] = `
<DocumentFragment>
  <form>
    <input
      type="text"
      value=""
    />
    <button>
      Save
    </button>
  </form>
</DocumentFragment>
`;
